jest.d.ts

```diff
@@ -0,0 +1,7 @@
+declare namespace jest {
+  interface Matchers<R> {
+    toHaveProp(selector: string): void;
+    toRender(selector: string): void;
+    toHaveState(selector: string, arg2: any): void;
```

Comment:
```suggestion
    toHaveState(selector: string, expected: unknown): R;
```

https://github.com/facebook/jest/blob/ccea646e76efd73654468ac32b3cb3a88778f7f4/packages/expect/src/types.ts#L205